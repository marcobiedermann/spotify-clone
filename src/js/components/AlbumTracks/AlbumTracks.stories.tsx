import type { Meta, StoryObj } from '@storybook/react';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import AlbumTracks from '.';
import albumTrackFixtures from '../AlbumTrack/__fixtures__';

const meta: Meta<typeof AlbumTracks> = {
  component: AlbumTracks,
  args: {
    items: [albumTrackFixtures],
  },
  decorators: [
    (Story) => (
      <MemoryRouter>
        <Story />
      </MemoryRouter>
    ),
  ],
};

type Story = StoryObj<typeof AlbumTracks>;

const Default: Story = {};

export { Default };
export default meta;
