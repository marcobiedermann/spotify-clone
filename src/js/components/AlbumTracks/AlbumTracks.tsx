import AlbumTrack, { AlbumTrackProps } from '../AlbumTrack';

export interface AlbumTracksProps {
  items: AlbumTrackProps[];
}

function AlbumTracks(props: AlbumTracksProps): JSX.Element {
  const { items } = props;

  return (
    <table>
      <thead>
        <tr>
          <th>#</th>
          <th>Title</th>
          <th align="right">Duration</th>
        </tr>
      </thead>
      <tbody>
        {items.map((item) => (
          <AlbumTrack key={item.id} {...item} />
        ))}
      </tbody>
    </table>
  );
}

export default AlbumTracks;
