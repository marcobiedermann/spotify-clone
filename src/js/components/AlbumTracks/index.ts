import AlbumTracks from './AlbumTracks';

export default AlbumTracks;
export * from './AlbumTracks';
