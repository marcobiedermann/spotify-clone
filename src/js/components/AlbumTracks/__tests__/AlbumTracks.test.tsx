import { render } from '@testing-library/react';
import React from 'react';
import { expect, it } from 'vitest';
import AlbumTracks from '..';

it('renders correctly', () => {
  const { container } = render(<AlbumTracks items={[]} />);

  expect(container.firstChild).toMatchSnapshot();
});
