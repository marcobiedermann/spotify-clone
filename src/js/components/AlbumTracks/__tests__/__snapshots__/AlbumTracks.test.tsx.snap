// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders correctly 1`] = `
<table>
  <thead>
    <tr>
      <th>
        #
      </th>
      <th>
        Title
      </th>
      <th
        align="right"
      >
        Duration
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        0
      </td>
      <td>
        <div
          class="track__name"
        >
          string
        </div>
        <div>
          <a
            href="/artists/string"
          >
            string
          </a>
        </div>
      </td>
      <td
        align="right"
      >
        0:00
      </td>
    </tr>
  </tbody>
</table>
`;
